src/lazy/encoder/writer.rs

```diff
@@ -35,6 +37,7 @@ pub(crate) struct WriterContext {
     #[allow(dead_code)]
     macro_table: MacroTable,
     num_pending_symbols: usize,
+    num_pending_macros: usize,
```

Comment:
🪧 The `Writer` now also tracks how many macros have been defined that haven't yet been written to output.